gems/aws-eventstream/lib/aws-eventstream/decoder.rb

```diff
@@ -63,26 +63,31 @@ class Decoder
       include Enumerable
 
       ONE_MEGABYTE = 1024 * 1024
+      private_constant :ONE_MEGABYTE
 
       # bytes of prelude part, including 4 bytes of
       # total message length, headers length and crc checksum of prelude
       PRELUDE_LENGTH = 12
+      private_constant :PRELUDE_LENGTH
 
       # bytes of total overhead in a message, including prelude
       # and 4 bytes total message crc checksum
-      OVERHEAD_LENGTH = 16
+      CRC32_LENGTH = 4
+      private_constant :CRC32_LENGTH
 
       # @param [Hash] options The initialization options.
       # @option options [Boolean] :format (true) When `false` it
       #   disables user-friendly formatting for message header values
       #   including timestamp and uuid etc.
       def initialize(options = {})
         @format = options.fetch(:format, true)
-        @message_buffer = BytesBuffer.new('')
+        @message_buffer = ''
       end
 
-      # @return [BytesBuffer]
-      attr_reader :message_buffer
+      # exposed for testing
+      def _message_buffer
```

Comment:
Similar comment here - leaves this attr_reader :message_buffer and change the documentation of return type.

Since this was exposed previously and isn't api private, we should keep it exposed.  It is changing return types so could still be breaking, but string should expose (mostly) the same methods as the ByteBuffer.